--- setup.py.orig	2008-09-05 06:13:42.000000000 +0800
+++ setup.py	2008-09-05 06:14:11.000000000 +0800
@@ -43,6 +43,7 @@
       author='Romain Brette, Dan Goodman',
       author_email='Romain.Brette at ens.fr',
       download_url='https://gforge.inria.fr/frs/?group_id=1069',
+      license='CeCILL-2',
       classifiers=[
         'Development Status :: 4 - Beta',
         'Intended Audience :: Science/Research',
@@ -52,4 +53,4 @@
         'Programming Language :: Python',
         'Topic :: Scientific/Engineering :: Bio-Informatics'
         ]
-      )
\ No newline at end of file
+      )
