--- fixsetup.py.orig	2008-09-17 07:20:13.000000000 +0800
+++ fixsetup.py	2008-09-17 07:45:19.000000000 +0800
@@ -0,0 +1,53 @@
+#!/usr/bin/python
+
+import glob
+import re
+import os
+import os.path
+
+todir = 'cocopy'
+
+pylist = glob.glob('*.py')
+del(pylist[pylist.index('setup.py')])
+del(pylist[pylist.index('fixsetup.py')])
+dirlist = ['documentation', 'examples', 'frames',
+           'pimaker', 'sources', 'testSuite']
+
+modlist = map(lambda py: os.path.splitext(py)[0], pylist)
+fixfiles = pylist[:]
+fixfiles.append('setup.py')
+fixfiles.extend(glob.glob(os.path.join('testSuite', '*.py')))
+for d in ['examples', 'frames', 'pimaker', 'sources']:
+    fixfiles.extend(glob.glob(os.path.join(d, '*.frame')))
+    fixfiles.extend(glob.glob(os.path.join(d, '*.atg')))
+
+restr = '(\\s*from\\s+)(' + '|'.join(modlist) + ')(\\s+import\\s+)'
+reobj = re.compile(restr)
+for ff in fixfiles:
+    ffp = open(ff)
+    output = ''
+    if ff == 'Coco.py': output = output + '#!/usr/bin/python\n'
+    ln = ffp.readline()
+    while ln:
+        output = output + reobj.sub('\\1cocopy.\\2\\3', ln)
+        ln = ffp.readline()
+    ffp.close()
+    ffp = open(ff, 'w')
+    ffp.write(output)
+    ffp.close()
+
+os.mkdir(todir)
+for f in pylist + dirlist:
+    os.rename(f, os.path.join(todir, f))
+
+if not os.path.isfile(os.path.join(todir, '__init__.py')):
+    inifile = open(os.path.join(todir, '__init__.py'), 'w')
+    inifile.write('\n')
+    inifile.close()
+
+if not os.path.isfile('Coco.py'):
+    scriptfile = open('Coco.py', 'w')
+    scriptfile.write('#!/usr/bin/python\n')
+    scriptfile.write('import sys; from cocopy.Coco import Coco\n')
+    scriptfile.write('Coco.main(sys.argv)\n')
+    scriptfile.close()
--- setupInfo.py.orig	2007-12-19 18:47:28.000000000 +0800
+++ setupInfo.py	2008-09-17 07:36:26.000000000 +0800
@@ -1,18 +1,23 @@
+import os.path
+
+name = 'CocoPy'
+version = '1.1.0rc'
+
 MetaData = {
-       'name':'CocoPy',
-       'version':'1.1.0rc',
+       'name':name,
+       'version':version,
        'description':'Python implementation of the famous CoCo/R LL(k) compiler generator.',
        'url':'http://www.ssw.uni-linz.ac.at/coco',
        'author':'Ron Longo',
        'author_email':'ron.longo@cox.net',
        'license':'GPL',
-       'packages':[ '' ],
-       'data_files':[ ( 'documentation', [ 'documentation/*' ] ),
-                      ( 'examples',      [ 'examples/*'      ] ),
-                      ( 'frames',        [ 'frames/*'        ] ),
-                      ( 'pimaker',       [ 'pimaker/*'       ] ),
-                      ( 'sources',       [ 'sources/*'       ] ),
-                      ( 'testSuite',     [ 'testSuite/*'     ] ) ],
+       'packages':[ 'cocopy' ],
+       'scripts':[ 'Coco.py' ],
+       'package_data':{'cocopy' : map(lambda dn: os.path.join(dn, '*.*'),
+                                       ['documentation', 'examples',
+                                        os.path.join('examples', 'Calc'),
+                                        'frames', 'pimaker',
+                                        'sources', 'testSuite'])},
        'classifiers':[
                       'Development Status :: 4 - Beta',
                       'Environment :: Console',
