--- setup.py.orig	2008-09-06 14:18:39.000000000 +0800
+++ setup.py	2008-09-06 14:19:02.000000000 +0800
@@ -9,7 +9,7 @@
 change, save, cascade and merge configurations. Interfaces easily with
 environment variables and command-line options. It has been developed on python
 2.3 but should work on version 2.2 or greater.""",
-   license="""Copyright (C) 2004-2007 by Vinay Sajip. All Rights Reserved. See LICENSE for license.""",
+   license="",
             version = "0.3.7",
             author = "Vinay Sajip",
             author_email = "vinay_sajip@red-dove.com",
