--- fixsetup.py.orig	2008-09-15 23:24:01.000000000 +0800
+++ fixsetup.py	2008-09-15 23:27:50.000000000 +0800
@@ -0,0 +1,47 @@
+#!/usr/bin/python
+
+import glob
+import re
+import os
+import os.path
+
+todir = 'cocopy'
+
+pylist = glob.glob('*.py')
+del(pylist[pylist.index('setup.py')])
+del(pylist[pylist.index('fixsetup.py')])
+
+modlist = []
+fixfiles = []
+os.mkdir(todir)
+for f in pylist:
+    modlist.append(os.path.splitext(f)[0])
+    tgtf = os.path.join(todir, f)
+    os.rename(f, tgtf)
+    fixfiles.append(tgtf)
+
+if not os.path.isfile(os.path.join(todir, '__init__.py')):
+    inifile = open(os.path.join(todir, '__init__.py'), 'w')
+    inifile.write('\n')
+    inifile.close()
+
+fixfiles.append('setup.py')
+fixfiles.extend(glob.glob(os.path.join('testSuite', '*.py')))
+for d in ['examples', 'frames', 'pimaker', 'sources']:
+    fixfiles.extend(glob.glob(os.path.join(d, '*.frame')))
+    fixfiles.extend(glob.glob(os.path.join(d, '*.atg')))
+
+restr = '(\\s*from\\s+)(' + '|'.join(modlist) + ')(\\s+import\\s+)'
+reobj = re.compile(restr)
+
+for ff in fixfiles:
+    ffp = open(ff)
+    output = ''
+    ln = ffp.readline()
+    while ln:
+        output = output + reobj.sub('\\1cocopy.\\2\\3', ln)
+        ln = ffp.readline()
+    ffp.close()
+    ffp = open(ff, 'w')
+    ffp.write(output)
+    ffp.close()
--- setupInfo.py.orig	2007-12-19 18:47:28.000000000 +0800
+++ setupInfo.py	2008-09-15 23:24:01.000000000 +0800
@@ -1,18 +1,33 @@
+import os
+import os.path
+
+def all_files(dirpath):
+    result = []
+    for entry in os.listdir(dirpath):
+        entrypath = os.path.join(dirpath, entry)
+        if os.path.isdir(entrypath): result.extend(all_files(entrypath))
+        elif os.path.isfile(entrypath): result.append(entrypath)
+    return result
+
+def data_dir(dirname):
+    return (os.path.join('share', '%s-%s' % (name, version), dirname),
+            all_files(dirname))
+
+name = 'CocoPy'
+version = '1.1.0rc'
+
 MetaData = {
-       'name':'CocoPy',
-       'version':'1.1.0rc',
+       'name':name,
+       'version':version,
        'description':'Python implementation of the famous CoCo/R LL(k) compiler generator.',
        'url':'http://www.ssw.uni-linz.ac.at/coco',
        'author':'Ron Longo',
        'author_email':'ron.longo@cox.net',
        'license':'GPL',
-       'packages':[ '' ],
-       'data_files':[ ( 'documentation', [ 'documentation/*' ] ),
-                      ( 'examples',      [ 'examples/*'      ] ),
-                      ( 'frames',        [ 'frames/*'        ] ),
-                      ( 'pimaker',       [ 'pimaker/*'       ] ),
-                      ( 'sources',       [ 'sources/*'       ] ),
-                      ( 'testSuite',     [ 'testSuite/*'     ] ) ],
+       'packages':[ 'cocopy' ],
+       'data_files':map(lambda dirname: data_dir(dirname), 
+                        ['documentation', 'examples', 'frames',
+                         'pimaker', 'sources', 'testSuite']),
        'classifiers':[
                       'Development Status :: 4 - Beta',
                       'Environment :: Console',
