--- setup.py.bk	2008-09-06 16:36:35.000000000 +0800
+++ setup.py	2008-09-06 16:36:49.000000000 +0800
@@ -33,7 +33,7 @@
     author = "Jeff Younker",
     author_email = "jeff@theblobshop.com",
     description = """A tool for performing reversible database migrations.""",
-    license = "Copyright: Shannon -jj Behrens <jjinux@gmail.com> -- distributed under the same license as pylons",
+    license = "BSD",
     keywords = "db dbapi sql migration upgrade database",
     url = "http://www.theblobshop.com/dbmigrate",
     download_url = "http://www.theblobshop.com/downloads/dbmigrate-%s.tar.gz" % version,
