--- setup.py.bk	2008-09-06 22:16:12.000000000 +0800
+++ setup.py	2008-09-06 22:16:49.000000000 +0800
@@ -35,13 +35,7 @@
     author_email='c@cthedot.de',
     url='http://cthedot.de/encutils/',
     download_url='http://cthedot.de/encutils/',
-    license='''encutils has a dual-license, please choose whatever you prefer:
-
-    * encutils is published under the `LGPL 3 or later <http://cthedot.de/encutils/license/>`__
-    * encutils is published under the
-      `Creative Commons License <http://creativecommons.org/licenses/by/3.0/>`__.
-
-    ''',
+    license='LGPL-3',
     keywords=['encoding', 'i18n', 'xml', 'html', 'css'],
     platforms='Python 2.3 and later.',
     classifiers=[
