--- setup.py.bk	2008-09-07 00:28:43.000000000 +0800
+++ setup.py	2008-09-07 00:30:18.000000000 +0800
@@ -18,7 +18,7 @@
 
 
 import os, sys, glob, fnmatch
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 import distutils.command.install_data
 
 import fuimodules.i18n # to install gettext for fuimodules.strings
@@ -77,7 +77,7 @@
     description = fuimodules.strings.description,
     author = "Donn.C.Ingle",
     author_email = fuimodules.strings.contact,
-    license = fuimodules.strings.licence,
+    license = 'GPL-3',
     url = fuimodules.strings.url,
     packages = ['fuimodules'],
     data_files = files,
